### PR TITLE
Edits For 2GP Package Creation

### PR DESCRIPTION
**Pull request by ShubhamVijaivargiya**
## Summary
2GP Package Creation with changed namespace, postInstall script automation, metadata visibility related changes/control edits, namedCredential reference edits, B2B Commerce dependency definition and B2B Apex Lib dependency definition, scratch org definition for package validation.

## Tested scenarios

**Fixed issue**:  PW-6316

### CODE REVIEW

**Review by wboereboom — APPROVED** [review]

File: config/project-scratch-def.json [hunk 1]
```diff
@@ -1,10 +1,23 @@
 {
-  "orgName": "harm Company",
+  "orgName": "shubhamv company",
```
Comment (maassenbas) [inline config/project-scratch-def.json:2]: Can we name this `adyen` instead?

File: force-app/main/default/customMetadata/Merchant.AdyenMerchantConfig.md-meta.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <CustomMetadata xmlns="http://soap.sforce.com/2006/04/metadata" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
     <label>AdyenMerchantConfiguration</label>
-    <protected>true</protected>
+    <protected>false</protected>
```
Comment (maassenbas) [inline force-app/main/default/customMetadata/Merchant.AdyenMerchantConfig.md-meta.xml:4]: Why updating it to `false`?

File: sfdx-project.json [hunk 3]
```diff
@@ -2,10 +2,29 @@
   "packageDirectories": [
     {
       "path": "force-app",
-      "default": true
+      "default": true,
+      "package": "Adyen-Plugin-For-SFB2B-VF",
```
Comment (maassenbas) [inline sfdx-project.json:6]: Let's keep consistency between names and call it `adyen-salesforce-b2b-commerce`

File: force-app/main/default/customMetadata/Merchant.AdyenMerchantConfig.md-meta.xml (hunk 2 above)
Comment (ShubhamVijaivargiya) [reply force-app/main/default/customMetadata/Merchant.AdyenMerchantConfig.md-meta.xml:4]: @maassenbas I will take care of this in my PR for PW-6399. This PR is for current versioned metadata(2.0.0).

File: sfdx-project.json (hunk 3 above)
Comment (ShubhamVijaivargiya) [reply sfdx-project.json:6]: This PR is for currently versioned metadata(2.0.0). I lost this name while testing previous versions, however, in a future release, I can explore possibilities of name change.

File: config/project-scratch-def.json (hunk 1 above)
Comment (ShubhamVijaivargiya) [reply config/project-scratch-def.json]: Done

**Pull request merged**